internal/resource/statefulset_test.go

```diff
@@ -1011,12 +1011,19 @@ var _ = Describe("StatefulSet", func() {
 			}))
 		})
 
-		It("adds the required terminationGracePeriodSeconds", func() {
+		It("sets TerminationGracePeriodSeconds in podTemplate as provided in instance spec", func() {
+			ten := int64(10)
+			instance.Spec.TerminationGracePeriodSeconds = &ten
```

Comment:
There are a couple of places in this PR where you could use the `"k8s.io/utils/pointer"` package if you want to as done in https://github.com/rabbitmq/cluster-operator/blob/01b368312c24625bb74efbbe5bd860534b9d43a1/internal/resource/statefulset.go#L524

Reply:
Good point. On it!